include/triton/common/logging.h

```diff
@@ -33,46 +33,54 @@
 #include <string>
 #include <vector>
 
-namespace triton { namespace common {
-
-// A log message.
-class LogMessage {
- public:
-  // Log levels.
-  enum Level { kERROR = 0, kWARNING = 1, kINFO = 2 };
+#include "table_printer.h"
+#ifdef _WIN32
+// suppress the min and max definitions in Windef.h.
+#define NOMINMAX
+#include <Windows.h>
+#else
+#include <sys/time.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+#endif
 
-  LogMessage(const char* file, int line, uint32_t level);
-  ~LogMessage();
 
-  std::stringstream& stream() { return stream_; }
+namespace triton { namespace common {
 
- private:
-  static const std::vector<char> level_name_;
-  std::stringstream stream_;
-};
 
 // Global logger for messages. Controls how log messages are reported.
 class Logger {
  public:
+  // Log Formats.
   enum class Format { kDEFAULT, kISO8601 };
 
+  // Log levels.
+  enum Level { kERROR = 0, kWARNING = 1, kINFO = 2, kEND };
```

Comment:
`enum class`?

Reply:
actually maybe not - let me check - we use these to index into the array of levels - and not 100% that will work with enum class

Reply:
yep - this needs to be enum and not enum class - enum enables conversion to unsigned int which we use to index

Reply:
Isn't that what `enum class Level : uint8_t` or something would do but with stronger type checking?

Reply:
Mostly a nit otherwise, I don't mind `enum` if the `enum class` change wouldn't be straightforward.

Reply:
oh man - I should have read this :) - enum class required casting in a few places - let me know if you have a preference